live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/domain/event/IngestNewNodeEventHandlerTest.java

```diff
@@ -28,41 +28,80 @@
 
 import static org.mockito.BDDMockito.given;
 import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
 
+import java.util.Optional;
+
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 
 import org.alfresco.hxi_connector.live_ingester.domain.model.in.IngestNewNodeEvent;
+import org.alfresco.hxi_connector.live_ingester.domain.model.in.Node;
 import org.alfresco.hxi_connector.live_ingester.domain.model.out.event.EventPublisher;
 import org.alfresco.hxi_connector.live_ingester.domain.model.out.event.UpdateNodeMetadataEvent;
+import org.alfresco.hxi_connector.live_ingester.domain.model.transform.request.TransformRequest;
+import org.alfresco.hxi_connector.live_ingester.domain.model.transform.request.TransformRequester;
 
 @ExtendWith(MockitoExtension.class)
 class IngestNewNodeEventHandlerTest
 {
     @Mock
     IngestNewNodeEvent ingestNewNodeEvent;
     @Mock
+    Node node;
+    @Mock
     UpdateNodeMetadataEvent updateNodeMetadataEvent;
     @Mock
+    TransformRequest transformRequest;
+    @Mock
     UpdateNodeEventMapper updateNodeEventMapper;
     @Mock
     EventPublisher eventPublisher;
+    @Mock
+    TransformRequestMapper transformRequestMapper;
+    @Mock
+    TransformRequester transformRequester;
     @InjectMocks
     IngestNewNodeEventHandler ingestNewNodeEventHandler;
 
+    @BeforeEach
+    void setUp()
+    {
+        given(ingestNewNodeEvent.node()).willReturn(node);
+    }
+
+    @Test
+    void nodeWithoutContent_PublishMetadataButNoTransform()
+    {
+        // given
+        given(node.contentMimeType()).willReturn(Optional.empty());
+        given(updateNodeEventMapper.map(ingestNewNodeEvent)).willReturn(updateNodeMetadataEvent);
+
+        // when
+        ingestNewNodeEventHandler.handle(ingestNewNodeEvent);
+
+        // then
+        verify(eventPublisher).publishMessage(updateNodeMetadataEvent);
```

Comment:
You are mixing here Mockito's normal syntax with BDD - I think we should stick to one - `when/verify` or `given/then`. I prefer the second (BDD) option. What do you prefer?

Reply:
I don't mind either way - they're both readable. The `then` syntax is slightly more verbose (especially when compared with `verifyNoInteractions(transformRequestMapper, transformRequester);`).

Reply:
Interestingly PMD complains when using `given/then` because it knows to expect `verify` in a test:
> JUnit tests should include assert() or fail()

Reply:
Oh! Didn't expect that 😅 Again there is space to improve PMD, but for now we could suppress the warning... or stick to `when/verify`. I like this BDD approach, it would be a pity to resign from it due to PMD

Reply:
I've raised a bug addressing this issue: https://github.com/pmd/pmd/issues/4781